music21/metadata/__init__.py

```diff
@@ -582,95 +1255,6 @@ def search(self, query=None, field=None, **kwargs):
                     return True, innerField
         return False, None
 
-    def setWorkId(self, idStr, value):
-        r'''
-        Directly set a work id, given either as a full string name or as a
-        three-character abbreviation. The following work id abbreviations and
-        their full id string are given as follows. In many cases the Metadata
-        object support properties for convenient access to these work ids.
-
-        Abbreviations and strings::
-            * otl / title
-            * otp / popularTitle
-            * ota / alternativeTitle
-            * opr / parentTitle
-            * oac / actNumber
-            * osc / sceneNumber
-            * omv / movementNumber
-            * omd / movementName
-            * ops / opusNumber
-            * onm / number
-            * ovm / volume
-            * ode / dedication
-            * oco / commission
-            * gtl / groupTitle
-            * gaw / associatedWork
-            * gco / collectionDesignation
-            * txo / textOriginalLanguage
-            * txl / textLanguage
-            * ocy / countryOfComposition
-            * opc / localeOfComposition.
-
-        >>> md = metadata.Metadata(title='Quartet')
-        >>> md.title
-        'Quartet'
-
-        >>> md.setWorkId('otl', 'Trio')
-        >>> md.title
-        'Trio'
-
-        >>> md.setWorkId('ocy', 'Latvia')
-        >>> md.ocy
-        'Latvia'
-        >>> md.countryOfComposition
-        'Latvia'
-
-        >>> md.setWorkId('sdf', None)
-        Traceback (most recent call last):
-        music21.exceptions21.MetadataException: no work id available with id: sdf
-        '''
-        idStr = idStr.lower()
-        match = False
-        for abbreviation, workId in self.workIdAbbreviationDict.items():
-            # for id in WORK_IDS:
-            # abbreviation = workIdToAbbreviation(id)
-            if workId.lower() == idStr:
-                self._workIds[workId] = Text(value)
-                match = True
-                break
-            elif abbreviation == idStr:
-                self._workIds[workId] = Text(value)
-                match = True
-                break
-        if not match:
-            raise exceptions21.MetadataException(
-                f'no work id available with id: {idStr}')
-
-    @staticmethod
-    def workIdToAbbreviation(value):
-        '''Get a work abbreviation from a string representation.
-
-        >>> metadata.Metadata.workIdToAbbreviation('localeOfComposition')
-        'opc'
-
-        Static method.
-        '''
-        # NOTE: this is a performance critical function
```

Comment:
we should check performance on the equivalent functions, since at some point this slowed down indexing of the whole corpus substantially.

Reply:
The equivalent functions are faster now (they're just dictionary lookups) BUT search has slowed down, since it is O(n) where n is the number of supported metadata terms (which has increased dramatically).  So test runs are definitely slower.  I will look to see if I can rework search to be O(m) where m is the number of items in the actual metadata for the score.

Reply:
That would be great -- or at least make an issue so we don't forget it.